#161962908 Retrieve and address users complaints

victor-nweze-deel opened the pull request:
#### What does this PR do?
Retrieve all complaints and addresses complaints

#### Description of Task to be completed?
- Create a column on the the complaints table for admin response
- Create routes to handle admin requests
- Write integration tests
- Document endpoints

#### How should this be manually tested?
- Git clone this repository
- Cd into the `haven-ah-backend` folder
- Run `npm run start-dev` to start the dev server
- Start up postman and hit the route `localhost:5000/api/v1/admin/users/complaints` with a `GET` request,
- To reply to a complaint hit the route `localhost:5000/api/v1/admin/users/complaints/:id/reply` with a `PUT` request
- To run the integration tests, run `npm test`

#### What are the relevant pivotal tracker stories?
[#161962908](https://www.pivotaltracker.com/story/show/161962908)

#### Screenshots (if appropriate)
<img width="1213" alt="screen shot 2018-11-16 at 8 56 17 am" src="https://user-images.githubusercontent.com/41582565/48614758-50cce300-e98f-11e8-881e-c67e7594c6dd.png">
<img width="1216" alt="screen shot 2018-11-16 at 9 02 15 am" src="https://user-images.githubusercontent.com/41582565/48614762-53c7d380-e98f-11e8-95d3-c5bf062ba0f3.png">

PriscillaSam committed:
feat(roles): add endpoint to update a user's role

- add countroller function for the endpoint
- test function and route with mocha and chai
- add swagger documentation for the endpoint

[Finishes #161924003]

TheoOkafor commented on server/migrations/20181115171611-update-complaint-model.js:
```diff
@@ -0,0 +1,16 @@
+      'Complaints',
+      'adminAction',
+      {
+        type: Sequelize.TEXT,
```
@ProfJigsaw Why are we using TEXT here instead of STRING dataType?

victor-nweze-deel replied:
The admins response could be anything from a simple `Thank you for your concern. We are processing your request` to a full text response dispelling a raised complaint.

TheoOkafor commented on server/repository/complaintRepository.js:
```diff
@@ -0,0 +1,52 @@
+    if (!complaint) {
+      return null;
+    }
+    return complaint.update({
```
Nice one.

TheoOkafor reviewed (commented):
This looks good. Nice work.

victor-nweze-deel committed:
ft(admin): Retrieve and respond to complaints

- Create a column on the the complaints table for admin response
- Create routes to handle admin requests
- Write integration tests
- Document endpoints

ucheg6 commented:
Good work @ProfJigsaw.
